### ISSUE TITLE
luks2john Python Implementation Does Not Support xts-plain64

### ISSUE THREAD

**Issue by wesdawg** [magnumripper/JohnTheRipper#2169]
The luks2john.py implementation does not seem to work on LUKS headers that aren't cbc-essiv:sha256 , such as xts-plain64:sha1. xts-plain64 is the default used by the more recent cryptsetup versions starting at 1.6.0 (https://security.stackexchange.com/questions/39306/how-secure-is-ubuntus-default-full-disk-encryption).

This would appear to be a regression from the functionality in the C code implementation since running the luks2john exe from 1.8.0-Jumbo-1 , I was able to produce a hash file from an xts-plain64:sha1 encrypted header.

**Comment by kholia** [comment]
This is not a regression. The generated hash could not be cracked by the format anyways. So such hashes aren't printed out now.

Patches are welcome to support  xts-plain64 in the LUKS format.

**Comment by wesdawg** [comment]
So to be clear, you're saying that xts-plain64 sha1 is not crackable with
the current build even though the hash and dump files are generated by the
1.8.0-Jumbo-1 luks2john ?
Just want to know so I don't waste my time running something that won't
work anyway.

Thanks,

**Comment by kholia** [comment]
> So to be clear, you're saying that xts-plain64 sha1 is not crackable with
> the current build even though the hash and dump files are generated by the
> 1.8.0-Jumbo-1 luks2john?
>
> Just want to know so I don't waste my time running something that won't
> work anyway.

Yes, xts-plain64:sha1 hashes are not crackable yet. The older luks2john.c generated such hashes wrongly.

**Comment by kholia** [comment]
On Tue, Jul 19, 2016 at 11:01 PM, wesdawg <notifications@github.com> wrote:
>
> So to be clear, you're saying that xts-plain64 sha1 is not crackable with
> the current build even though the hash and dump files are generated by the
> 1.8.0-Jumbo-1 luks2john?
>
> Just want to know so I don't waste my time running something that won't
> work anyway.

Yes, xts-plain64:sha1 hashes are not crackable yet. The older
luks2john.c generated such hashes wrongly.

> On Mon, Jul 11, 2016 at 4:22 AM, Dhiru Kholia <notifications@github.com>
> wrote:
>
> > This is not a regression. The generated hash could not be cracked by the
> > format anyways. So such hashes aren't printed out now.
> >
> > Patches are welcome to support xts-plain64 in the LUKS format.

**Comment by kholia** [comment]
LUKS2 recognition / support would also be nice.

https://mbroz.fedorapeople.org/talks/DevConf2016/devconf2016-luks2.pdf
